### PR TITLE
Python Ruff support (config only)

### PR DESCRIPTION
**Pull request by aaronsteers**
This can safely merge ahead of the large auto-fix PR:

- https://github.com/airbytehq/airbyte/pull/29853

This PR adds configuration for Ruff, allowing developers to applying their own auto-fixes and to see what changes or issues Ruff would find, without making it a breaking change and without impacting any existing workstreams.

## To test in VS Code

Install the VS Code extension for Ruff: https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff

## To test with the command line

Install Ruff with pip or pipx:

```console
pipx install ruff
```

Run Ruff on the entire monorepo:

```console
ruff check --fix .
```

OR run Ruff on any subproject's directory:

```console
ruff check --fix airbyte-ci/connectors/pipelines
ruff check --fix airbyte-cdk/python
```

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -1 +0,0 @@
-3.9.11
```
Comment (natikgadzhi) [inline .python-version:1]: Why is this deleted? Should we set Python 3.10 here? I think we use that in most cases and it's compatible with airbyte-ci, the CDK, and connectors.

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 140
-target-version = ["py37"]
+target-version = ["py38"]
```
Comment (natikgadzhi) [inline pyproject.toml:3]:
```suggestion
target-version = ["py310"]
```

File: pyproject.toml [hunk 3]
```diff
@@ -50,6 +50,94 @@ force_grid_wrap = 0
 use_parentheses = true
 skip_glob = ["**/connector_builder/generated/**"]
 
+[tool.ruff]
+target-version = "py38"
```
Comment (natikgadzhi) [inline pyproject.toml:54]:
```suggestion
target-version = "py310"
```

**Review by natikgadzhi — APPROVED** [review]
Approved with gotchas for Python 3.10 bump.

This also seems to be relatively safe (although I haven't tested, perhaps this would change the format that black outputs and then we would need to follow-up with a formatting fix). 

Let's get this in, @aaronsteers!

**Comment by natikgadzhi** [comment]
I've merged master into this, and added ruff as explicit dev dependency at the root pyproject.toml.

File: pyproject.toml [hunk 4]
```diff
@@ -46,11 +47,11 @@ max-complexity = 20
 max-line-length = 140
 
 extend-ignore = [
```
Comment (aaronsteers) [inline pyproject.toml:49]: Just formatting noise here. I'll revert.

Comment (natikgadzhi) [reply pyproject.toml:49]: Why tho, let it. Whatever, not terrible.

File: pyproject.toml [hunk 5]
```diff
@@ -63,11 +64,99 @@ include_trailing_comma = true
...
+[tool.ruff.flake8-pytest-style]
+parametrize-values-type = "tuple"
+
+[tool.ruff.isort]
+known-first-party = [
+    "airbyte",
+    "airbyte_cdk",
+    "airbyte_protocol",
+    "airbyte_api_client",
+    "connector_ops",
```
Comment (natikgadzhi) [inline pyproject.toml:152]: Ironically this is missing a bunch of those airbyte-ci packages, but I think this is alright. This is enough to start.

**Pull request merged**